crypto/evp/pmeth_fn.c

```diff
@@ -38,7 +38,7 @@ static int evp_pkey_asym_cipher_init(EVP_PKEY_CTX *ctx, int operation)
      */
     ERR_set_mark();
 
-    if (ctx->engine != NULL)
+    if (ctx->keytype != NULL)
```

Comment:
~~You **sure** about this?~~

Please revert this change. Everything else looks good, and with this one reverted all the OpenSSL and all the relevant `libp11` tests pass.

Reply:
Specifically this one, *only* checking 'ctx->engine`, was plain wrong.

Reply:
D'oh!  `!=` should be `==`...